notebooks/COS/ViewData/ViewData.ipynb

```diff
@@ -0,0 +1,1452 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize = (10,6))\n",
+    "for i in range(2):\n",
+    "    wvln, flux, fluxErr, dataQual, segment = fuv_x1d_data[i][\"WAVELENGTH\"], fuv_x1d_data[i][\"FLUX\"],\\\n",
+    "    fuv_x1d_data[i][\"ERROR\"], fuv_x1d_data[i][\"DQ\"], fuv_x1d_data[i][\"SEGMENT\"]\n",
+    "\n",
+    "\n",
+    "    plt.plot(wvln[dataQual != 128.0], flux[dataQual != 128.0], linewidth = 1, alpha = 0.5, \n",
+    "             label = f\"{segment} Cleaned\")\n",
+    "    plt.scatter(wvln[dataQual == 128.0], flux[dataQual == 128.0], s = 2, c = 'r', alpha = 1, \n",
+    "                label = f\"{segment} Pix out-of-bounds\")\n",
+    "plt.legend(fontsize = 18)\n",
+    "plt.title(\"Fig. 3.3\\nRemoving only out-of-bounds pixels from the FUV\\n\", size = 25)\n",
+    "plt.xlabel('Wavelength [$\\AA$]', size = 20)\n",
+    "plt.ylabel('Flux [$erg\\ s^{-1}\\ cm^{-2}\\ Angstrom^{-1}$]', size = 15)\n",
+    "plt.tight_layout()\n",
+    "plt.savefig(plotsdir+'Fig3.3.png', dpi = 200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise 3.1: *Removing all flagged data from the NUV*\n",
+    "\n",
+    "The NUV data has flags for more than just out-of-bounds pixels.\n",
+    "\n",
+    "1. Filter out all data which has been flagged\n",
+    "2. Plot the NUV spectrum from stripes NUVA and NUVB\n",
+    "3. Plot the flagged data in another color to see where the removed data is"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Your answer here\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=BinningV></a>\n",
+    "## 3.2. Binning \n",
+    "\n",
+    "Up until now, the data we've been working with has one value per pixel of the detector.\n",
+    "COS data is generally binned to a resolution element, or [resel](https://hst-docs.stsci.edu/cosihb/chapter-3-description-and-performance-of-the-cos-optics/3-2-size-of-a-resolution-element), when working with it. These resels are *generally* the following sizes *in pixels*.\n",
+    "\n",
+    "||FUV|NUV|\n",
+    "|-|---|---|\n",
+    "|**Dispersion Axis**|6|3|\n",
+    "|**Cross-Dispersion Axis**|10|3|\n",
+    "\n",
+    "So, for the FUV, we generally want to preliminarily bin our x1d or x1dsum data by 6.\n",
+    "\n",
+    "In the next few cells below, we define two functions:\n",
+    "1. `downsample_1d`, for binning an array \n",
+    "2. `binByResel`, which applies this binning to an entire COS NUV or FUV dataset\n",
+    "\n",
+    "<a id=BinfunV></a>\n",
+    "### 3.2.1. Creating Some Useful Functions for Binning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from Scripts.Useful_COS_Functions import downsample_1d"
```

Comment:
You'll have to make some changes elsewhere to implement this of course, but the python convention for packages is:
```suggestion
    "from scripts.useful_COS_functions import downsample_1d"
```
i.e., module names should be lower-case separated by undersocres (and "COS" could be "cos", but there's not really agreement on whether acronyms should follow that rule so :shrug:)